Convert GenomicFiles.Rnw to GenomicFiles.Rmd

sonali8434 opened the pull request:
I am working on this code of Biconductor/GenomicFiles .

sonali8434 committed:
Convert file to Rmd

sonali8434 committed:
Remove file.Rnw

sonali8434 committed:
Update DESCRIPTION with BiocStyle, knitr, VignetteBuilder

sonali8434 commented:
@Bioconductor/sweave2rmd
I have completed  the 1st  task and also converted the GenomicFiles.Rnw to GenomicFiles.Rmd please review my pull request .
Replacing_GenomicFiles.Rnw_to_GenomicFiles_Rmd 
Thank You!

sonali8434 committed:
Update GenomicFiles.Rmd

jwokaty commented on DESCRIPTION:
```diff
@@ -10,14 +10,18 @@ Authors@R: c(person("Bioconductor Package Maintainer", role = c("aut", "cre"),
-	      person("Martin", "Morgan", role = "aut"))
+	      person("Martin", "Morgan", role = "aut"),
+              person("Sonali", "Kumari", role = "ctb",
+              comment = "Converted GinomicFles vignette from GenomicFiles.Rnw to RMarkdown / HTML."),
```
Indent line 15. It should be indented like line 9.

Correct spelling of `GenomicFiles`. We typically use the following for the comment: `Converted 'GenomicFiles' vignettes from Sweave to RMarkdown / HTML.`

sonali8434 replied:
ohk

jwokaty commented on DESCRIPTION:
```diff
@@ -10,14 +10,18 @@ Authors@R: c(person("Bioconductor Package Maintainer", role = c("aut", "cre"),
+              person("Sonali", "Kumari", role = "ctb",
+              comment = "Converted GinomicFles vignette from GenomicFiles.Rnw to RMarkdown / HTML."),
+        person("Bioconductor Package Maintainer",
+        email = "maintainer@bioconductor.org", role = "cre"))
```
Line 17 should be indented like line 9.

jwokaty commented on DESCRIPTION:
```diff
@@ -10,14 +10,18 @@ Authors@R: c(person("Bioconductor Package Maintainer", role = c("aut", "cre"),
 Imports: GenomicAlignments (>= 1.7.7), IRanges, S4Vectors (>= 0.9.25),
         VariantAnnotation (>= 1.27.9), GenomeInfoDb
-Suggests: BiocStyle, RUnit, genefilter, deepSNV, snpStats,
+Suggests: BiocStyle, RUnit, genefilter, deepSNV, snpStats,BiocStyle,knitr
```
BiocStyle already exists in Suggests. Please remove the additional BiocStyle. Leave `knitr` but add a space before the comma.

sonali8434 replied:
I have made all the changes

sonali8434 committed:
Update DESCRIPTION

sonali8434 committed:
Update DESCRIPTION

sonali8434 committed:
Update DESCRIPTION

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,703 @@
+---
+editor_options: 
+  markdown: 
+    wrap: 72
+---
+
+---
+title: "GenomicFiles"
+author: Valerie Obenchain,Michael Love,Lori Shepherd,Martin Morgan
+date: March 04, 2023
+package: GenomicFiles
+vignette: >
+  %\VignetteIndexEntry{GenomicFiles}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+output:
+  BiocStyle::html_document
+  
+---
```
- [x] Removing lines 1-6 because we don't need to wrap lines at 72 characters. There should be one YAML header.
- [x] The title of this file should be "Introduction to Genomic Files". This is what you see in https://bioconductor.org/packages/3.17/bioc/vignettes/GenomicFiles/inst/doc/GenomicFiles.pdf.
- [x] List the authors as follows
  ```
  author:
    - Valerie Obenchain
    - Michael Love
    - Lori Shepherd
    - Martin Morgan
  ```
- [x] You can add yourself as a contributor here with
  ```
    - name: Your Name
      affiliation: "Vignette translation from Sweave to R Markdown / HTML"
   ```
  Be careful with the spacing.
- [x] Rather than hard code the date, we use the following code so that it's always "current" when the file is regenerated:
  ```
  date: "`r format(Sys.time(), '%B %d, %Y')`"
  ```
  Since the PDF uses a "Last Modified" and a "Compiled Date", you can achieve this by using the code as the compiled date then just put last modified in front of it as in
  ```
  date: "Last Modified October 2014"; "Compiled: `r format(Sys.time(), '%B %d, %Y')`"
  ```
- [x] On line 13, the VignetteIndexEntry should be `Introduction to GenomicFiles`. It's typically the same as the title.
- [x] Remove the extra space between `BiocStyle: html_document` and the `---`.

sonali8434 committed:
Update GenomicFiles.Rmd

sonali8434 committed:
Update GenomicFiles.Rmd

jwokaty commented on DESCRIPTION:
```diff
@@ -10,14 +10,18 @@ Authors@R: c(person("Bioconductor Package Maintainer", role = c("aut", "cre"),
+	      person("Martin", "Morgan", role = "aut"),
+              person("Sonali", "Kumari", role = "ctb",
+                     comment = "Converted 'GenomicFiles' vignettes from Sweave to RMarkdown / HTML."),
+        person("Bioconductor Package Maintainer",
```
Please indent line 16. It should be the same indentation as lines 13 and 14

sonali8434 replied:
Done !!

sonali8434 committed:
Update GenomicFiles.Rmd

sonali8434 committed:
Update DESCRIPTION

sonali8434 committed:
Update GenomicFiles.Rmd

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,705 @@
+
+This vignette illustrates how to use the `GenomicFiles` package for distributed computing across files. The functions in `GenomicFiles` manipulate and combine data subsets via two user-supplied functions, `MAP` and `REDUCE`. These are similar in spirit to `Map` and `Reduce` in base R. Together they provide a flexible interface to extract, manipulate, and combine data. Both functions are executed  ...
+
+We assume the reader has some previous experience with R and with basic manipulation of ranges objects such as `GRanges` and `GAlignments` and file classes such as `BamFile` and `BigWigFile`. See the vignettes and documentation in `GenomicRanges`, `GenomicAlignments`, `Rsamtools` and `rtracklayer` for an introduction to these classes.
```
Please read https://bioconductor.org/packages/release/bioc/vignettes/BiocStyle/inst/doc/AuthoringRmdVignettes.html to learn how to make canonical links to packages using BiocStyle. GenomicRanges, GenomicAlignments, Rsamtools, and rtracklayer are all Bioconductor packages so they should be linked to. Do this for other packages you find in the vignette as well.

sonali8434 replied:
ohk !! Thanku

jwokaty replied:
I want to clarify the links a bit more because while you were able to generate the links, there's a better way to do it and  you won't see the effect until your vignette has `BiocStyle` included in the yaml yeader.

We use the functions mentioned in the BiocStyle documentation to generate the links. We do this because paths can later change, vignettes can always be knitted again, etc. If you look at Style Macros in BiocStyle, it says that you should use

     `r Biocpkg("GenomicFiles")` 

if you want knitr to generate a link in the HTML document. Please make this change for all Bioconductor packages in the vignette.

sonali8434 committed:
Update GenomicFiles.Rmd

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,712 @@
+contributor:
+  - name: Sonali Kumari
+    affiliation: "Vignette translation from Sweave to R Markdown / HTML"
+date: "`r format(Sys.time(), '%B %d, %Y')`"
```
I updated my note about this in my original comment since the original pdf has a last modified and a compiled date.

sonali8434 replied:
changed

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,712 @@
+  - Michael Love
+  - Lori Shepherd
+  - Martin Morgan
+contributor:
```
Remove line 8. There isn't a separate section for contributors.

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,712 @@
+
```
Remove space

sonali8434 committed:
Update GenomicFiles.Rmd

sonali8434 committed:
Update GenomicFiles.Rmd

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,712 @@
+
+The `GenomicFiles` package is available at bioconductor.org and can be downloaded via `BiocManager::install`. 
+
+```
+> if (!require("BiocManager"))
++ install.packages("BiocManager")
+> BiocManager::install("GenomicFiles")
+```
````
These should be R code blocks. (See Embed codes in https://www.rstudio.com/wp-content/uploads/2015/02/rmarkdown-cheatsheet.pdf). Typically, there is also a label so it can be identified with some attributes so that the code can be run (or not run). If you look at the corresponding .Rnw file, this part should be labeled `install` with `eval=FALSE` (https://github.com/Bioconductor/GenomicFiles/pull/8/files#diff-f1f895a7529de1d818ab65ace19caa98ab74c894fee01c0fe7060e57d30697c6L46-L50). It should look like

    ```{r install, eval=FALSE}
    if (!require("BiocManager"))
      install.packages("BiocManager")
    BiocManager::install("GenomicFiles")
    ```
    
I see all your code blocks don't have labels and they are not R code blocks but they should be. Please fix this code block and others in this file. Also take care of the spacing around the code blocks. There should be a blank line before and after a code block.

sonali8434 replied:
Ohk .. I will do the changes and let you know after getting all the things done.. For now you can review other's codes

jwokaty replied:
The code blocks look better.

jwokaty commented:
When you are finished updating the code blocks, please ask me to continue the review. I would also recommend getting into the habit of writing concise commit messages in case you ever need to hunt for a specific commit.

sonali8434 committed:
Update GenomicFiles.Rmd

jwokaty commented:
Can I ask why you are closing this?

sonali8434 commented:
> Can I ask why you are closing this?

Actually I have started working on this file in my PC(System) but i started getting errors while switching to this branch again in vscode . But now I have Successfully switched in this branch and also started working will let you know as soon as i complete the task . This time I will try to complete my task with minimum no of errors. 
Thank you !

sonali8434 commented:
I am Still in the process of learning that's why . But this time i will make Sure that i will complete it with accuracy

jwokaty commented:
It's okay. I know you're learning and I am here to help. You can always be proactive and explain what's going on. Communication is really important.

sonali8434 commented:
Currently I am learning how to make canonical links to packages using BiocStyle . And also making Relevant changes to the file .

sonali8434 committed:
Update GenomicFiles.Rmd

Update all the code chunks and reviewed myself from the pdf.

sonali8434 committed:
Corrected all the line spacing

sonali8434 commented:
> When you are finished updating the code blocks, please ask me to continue the review. I would also recommend getting into the habit of writing concise commit messages in case you ever need to hunt for a specific commit.

I am done with  updating the code blocks , you can now continue with the review . Thank You!

sonali8434 committed:
Update GenomicFiles.Rmd

Updated the file by Adding labels to each R code block

sonali8434 committed:
Update GenomicFiles.Rmd

Removed some of the label such as hide, eval=false , tex (getting errors while knitting)

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,691 @@
+---
+title: "Introduction to GenomicFiles"
+authors:
+  - name: "Valerie Obenchain"
+  - name: "Michael Love"
+  - name: "Shepherd"
+  - name: "Martin Morgan"
```
In addition to you, the only authors that should be listed here should be the ones in the PDF: https://bioconductor.org/packages/3.17/bioc/vignettes/GenomicFiles/inst/doc/GenomicFiles.pdf

sonali8434 committed:
Update DESCRIPTION

Removed the author name which is not present in pdf

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,691 @@
+    affiliation: "Vignette translation from Sweave to Rmarkdown / HTML"
+date: "`r format(Sys.time(), '%B %d, %Y')`"
+output:
+  html_document:
```
This line should read: `BiocStyle::html_document:`

sonali8434 replied:
Done

sonali8434 committed:
Update GenomicFiles.Rmd

removed the author name which is not present in pdf

jwokaty commented:
There are  few items that are still remaining from our previous round. Please address those changes.

We'll be using the following checklist to review your PR:

- [x] the `.Rmd` file knits to `HTML`
- [x] `R CMD build` runs without errors or timeouts
- [x] the tarball from `R CMD build` contains the `HTML` (check with the
  following by substituting the package name and vignette name `tar ztf
  package_name.tar.gz | grep 'doc/vignette_name'`)
- [x] the `.Rnw` file has been removed
- [x] in the `DESCRIPTION`, `BiocStyle` and `knitr` are listed in `Suggests`
- [x] in the `DESCRIPITON`, the line `VignetteBuilder: knitr` exists
- [x] any added lines use the same spacing and indents as the existing
  document
- [x] if agreed with the Maintainer, the contributor is in the author list
  in the `DESCRIPTION` file; refer to the contribution guide for acceptable
  formats
- [x] if agreed with the Maintainer, the contributor is in the author list
  in the vignette's YAML; refer to the contribution guide for acceptable formats
- [x] if this pull request includes a conversion from using the separate `Author`
  and `Maintainer` lines into the `Authors@R` vector, check that the
  `Authors@R` vector includes the maintainer as specified with `role='cre'`.
- [x] if this pull request includes a conversion from using the separate `Author`
  and `Maintainer` lines into the `Authors@R` vector, the `Maintainer` line
  is completely removed.
- [x] `HTML` document is representative of the `PDF` in content and
  in general the presentation
- [x] Where the contributor was not able to preserve the content and presentation
  of the `PDF` is noted as a comment in the pull request
- [x] the R Markdown file is representative of the Sweave document and follows
  best practices, such as replacing links to Bioconductor packages with calls
  to BiocStyle's `Biocpkg()`

jwokaty commented on DESCRIPTION:
```diff
@@ -9,15 +9,18 @@ Authors@R: c(person("Bioconductor Package Maintainer", role = c("aut", "cre"),
+	      person("Martin", "Morgan", role = "aut"),
+              person("Sonali", "Kumari", role = "ctb",
+                     comment = "Converted 'GenomicFiles' vignettes from Sweave to RMarkdown / HTML."),
+              person("Bioconductor Package Maintainer",
+                     email = "maintainer@bioconductor.org", role = "cre"))
```
I think these lines are duplicating the information in 8-9. Let's remove these lines, but remember to replace the comma on line 14 with a `)`.

sonali8434 replied:
Done

sonali8434 committed:
Update DESCRIPTION

Removed the line 15 and 16 since these lines are duplicating the information in 8-9.

jwokaty commented on DESCRIPTION:
```diff
@@ -9,15 +9,16 @@ Authors@R: c(person("Bioconductor Package Maintainer", role = c("aut", "cre"),
 		     email = "maintainer@bioconductor.org"),
 	      person("Valerie", "Obenchain", role = "aut"),
 	      person("Michael", "Love", role = "aut"),
-	      person("Lori", "Shepherd", role = "aut"),
```
Lori disappeared from this list.

sonali8434 replied:
resolved

sonali8434 committed:
Update DESCRIPTION

Added Lori

sonali8434 committed:
Update GenomicFiles.Rmd

BiocStyle: html_document added

sonali8434 committed:
Update GenomicFiles.Rmd

Updated introduction with linked packages using BiocStyle

sonali8434 committed:
Update GenomicFiles.Rmd

Adding 3 code blocks before Introduction for avoiding errors during knit

sonali8434 committed:
Update GenomicFiles.Rmd

Made all the required changes !! 
HTML file for Reference
HTML file link - https://rpubs.com/Sonali8434/1014072

sonali8434 commented:
Made all the required changes !! 
HTML file link - https://rpubs.com/Sonali8434/1014072
@jwokaty   You can continue with the review..

sonali8434 committed:
Update GenomicFiles.Rmd

Removed one code error {r} from introduction.

sonali8434 committed:
Update GenomicFiles.Rmd

Changed the date format

sonali8434 committed:
Update GenomicFiles.Rmd

Changed the date format

sonali8434 committed:
Update GenomicFiles.Rmd

Removed all the unnecessary library functions

sonali8434 commented:
Please start Review from here, This if final file after self reviewing and publishing my file.

sonali8434 committed:
Update GenomicFiles.Rmd

Added VignetteDepends in YAML

sonali8434 committed:
Update GenomicFiles.Rmd

Removed extra spaces

sonali8434 committed:
Update GenomicFiles.Rmd

Removed space from last line

sonali8434 committed:
Update GenomicFiles.Rmd

Checked every word spacing and spelling from the original pdf and corrected if any mistake is found

sonali8434 commented:
@jwokaty  While checking the spelling and spacing from the original pdf https://bioconductor.org/packages/devel/bioc/html/GenomicFiles.html . I have found out that there is one spelling mistake in original pdf in coverage 2 section in paragraph 2 there is written  ''ranges at at time.'' it should be ranges at a time . I just thought that i should inform you . 
Thank you .. 
I have checked each word from the pdf you can continue with the review process.

sonali8434 committed:
Update DESCRIPTION

sonali8434 committed:
Update GenomicFiles.Rmd

Removed extra space from pagebreak and also removed the extra name which was not in pdf in author section

sonali8434 commented:
@jwokaty Please Review my PR any Suggestion is highly appreciated.
Thank you

sonali8434 committed:
Update GenomicFiles.Rmd

Added space between the photo and a text

sonali8434 commented:
@jwokaty Please review my PR whenever you get time so that i can move forward with my work any suggestions will be appreciated.
Thank you..

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,687 @@
+  - name: "Sonali Kumari"
+    affiliation: "Vignette translation from Sweave to Rmarkdown / HTML"
+date: "Last modified: October 2014; Compiled: `r format(Sys.time(), '%B %d , %Y')`"
+
```
Remove space

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,687 @@
+  BiocStyle::html_document
+---
+
+```{r echo=FALSE, include=FALSE}
+if (!require("BiocManager", quietly = TRUE))
+    install.packages("BiocManager")
+BiocManager::install("BiocStyle", force = TRUE)
+```
````
Remove 20-24. If you read the Bioconductor developer documentation, it states that no package should automatically install packages. This is to give the user full control of their system, which is why all dependencies are documented in the DESCRIPTION file. Remove any extra blank lines. (Generally, we want 1 blank line surround paragraphs or code chunks for readability.)

sonali8434 replied:
well noted. Thanks for always giving valuable advice.

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,687 @@
+library(knitr)
+opts_chunk$set(tidy=FALSE)
+```
+
+```{r include=FALSE}
+library(BiocStyle)
+```
````
Remove 30-33. This shouldn't be needed and lines 16-17 should include BiocStyle.

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,687 @@
+We assume the reader has some previous experience with R and with basic
+manipulation of ranges objects such as `GRanges` and `GAlignments` and
+file classes such as `BamFile` and `BigWigFile`. See the vignettes and
+documentation in `r Biocpkg("GenomicRanges")`, `r Biocpkg("GenomicAlignments")`, `r Biocpkg ("GenomicAlignments")`, `r Biocpkg("Rsamtools")` and `r Biocpkg("rtracklayer")` for an introduction to these classes.
```
Wrap long lines such as these. We asked you how to do this in the Knowledge Quiz.

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,687 @@
+file classes such as `BamFile` and `BigWigFile`. See the vignettes and
+documentation in `r Biocpkg("GenomicRanges")`, `r Biocpkg("GenomicAlignments")`, `r Biocpkg ("GenomicAlignments")`, `r Biocpkg("Rsamtools")` and `r Biocpkg("rtracklayer")` for an introduction to these classes.
+
+The GenomicFiles package is available at bioconductor.org and can be downloaded via `BiocManager::install`.
+```{r install, eval=FALSE}
````
Please put a blank line in between paragraphs and code chunks. Do this for other instances in this document. Remember the goal is not just to convert the documents but to make them easier to maintain.

sonali8434 replied:
Well noted!

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,687 @@
+
+```{r quick_start-load, message=FALSE}
+library(GenomicFiles)
+
````
Remove blank lines at the beginning or end of code chunks.

sonali8434 replied:
> Hi @sonali8434 , your .Rmd is starting to look good and I am able to knit it. I am going to ask you to make a few more changes before I compare the new and old documents.
> 
> Good catch on the spelling mistake. Usually, if it is simple, it is okay to make. For others that are more complicated or where there are several, I usually advise the contributor to bring it up with reviewer/maintainer.

Hi @jwokaty I have done all the changes as suggested you can now continue with the review process.

sonali8434 replied:
> Hi @sonali8434 , your .Rmd is starting to look good and I am able to knit it. I am going to ask you to make a few more changes before I compare the new and old documents.
> 
> Good catch on the spelling mistake. Usually, if it is simple, it is okay to make. For others that are more complicated or where there are several, I usually advise the contributor to bring it up with reviewer/maintainer.

Hi @jwokaty I have done all the changes as suggested you can now continue with the review process.

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,687 @@
+ranges for each file then performs a basepair-level t-test across files.
+The t-test example also demonstrates how to use a blocking factor to
+differentiate files by experimental group (e.g., case vs control).
+\pagebreak
```
Remove `\pagebreak` plus any other instances of this code.

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,687 @@
+
+The `bam2R` function from the deepSNV package is used to compute the
+statistics. The MAP invokes `bam2R` and retains only the nucleotide
+counts (see ? `bam2R` for other output fields). Counts from the reference
```
Let's write this with no space in between as in 

    ?`bam2R`

This is because putting a question mark in front of a function's name in the console will display its documentation.

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,687 @@
+
+Define ranges of interest,
+```{r ttest-ranges, warning=FALSE}
+roi <- GRanges("chr14", IRanges(c(19411677, 19659063, 105421963,                        105613740), width=20))
````
Decrease the space between the last 2 ranges

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,687 @@
+    Values :    0   22   23   19   17   18   17 ...   20   22   21   23   22    0
+```
+
+\newpage
````
Remove `\newpage`

jwokaty commented:
Hi @sonali8434 , your .Rmd is starting to look good and I am able to knit it. I am going to ask you to make a few more changes before I compare the new and old documents.

Good catch on the spelling mistake. Usually, if it is simple, it is okay to make. For others that are more complicated or where there are several, I usually advise the contributor to bring it up with reviewer/maintainer.

sonali8434 committed:
Update GenomicFiles.Rmd

Done all the changes as suggested.

sonali8434 commented:
Hi @jwokaty I have done all the changes as suggested you can now continue with the review process.

sonali8434 committed:
Update GenomicFiles.Rmd

Removed knit lines because it was not in original document

sonali8434 committed:
Update GenomicFiles.Rmd

Added wrapping function in some more lines to improve code readability.

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,727 @@
+We assume the reader has some previous experience with R and with basic
+manipulation of ranges objects such as `GRanges` and `GAlignments` and file
+classes such as `BamFile` and `BigWigFile`. See the vignettes and documentation
+in `r Biocpkg("GenomicRanges")`, `r Biocpkg("GenomicAlignments")`, `r Biocpkg
+("GenomicAlignments")`, `r Biocpkg("Rsamtools")` and `r Biocpkg("rtracklayer")`
```
I believe GenomicAlignments is duplicated here. Remove the second one that is broken up at the end of line 32 and at the start of line 33.

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,727 @@
+("GenomicAlignments")`, `r Biocpkg("Rsamtools")` and `r Biocpkg("rtracklayer")`
+for an introduction to these classes.
+
+The GenomicFiles package is available at bioconductor.org and can be downloaded via `BiocManager::install`.
```
Wrap long line

sonali8434 replied:
I think i have done that already. but i will check once again

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,727 @@
+across files.
+
+```{r quick_start-MAP-REDUCE-reduceByRange}
+library(GenomicAlignments)
````
Remove line 98. This is not in the original PDF.

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,727 @@
+are sent to workers along with MAP and REDUCE functions. The MAP extracts and/or
+manipulates data and REDUCE consolidates the results from MAP. Both MAP and
+REDUCE are executed in the distributed step and therefore reduction occurs on
+data from the same worker, not across workers. The chart in Figure
```
Insert a line break between the end and start of these two sentences. These should be 2 paragraphs.

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,727 @@
+output of all MAP calls. `REDUCE` usually plays a minor role by concatenating or
+unlisting results.
+
+```{r echo = FALSE, fig1, fig.cap=": Mechanics of `reduceByRange` and `reduceRanges`"}
````
Remove the colon and space at the beginning of the caption. Also, usually the name of the code block comes  first, so it should read

    ````{r fig1, echo = FALSE, fig.cap = "Mechanics ..."}

I would also be consistent about little things, such as spacing around `=`.

sonali8434 replied:
Ohk well noted i will sit today and remove these mistakes carefully.

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,727 @@
+manipulates data and REDUCE consolidates the results from MAP. Both MAP and
+REDUCE are executed in the distributed step and therefore reduction occurs on
+data from the same worker, not across workers. The chart in Figure
+[1](reduceByRange_flow.png) represents the division of labor in `reduceByRange`
```
Rather than link directly to the image, it would be better to link to the place of image in the document. I think it is okay to use an anchor with id before the figure `<a id="fig1"></a>` then I think you can do `[1](#fig1)`. Please do this for any other images in this document.

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,727 @@
+```
+
+In contrast to the 'byRange' approach, `reduceByFile` and `reduceFiles` (Figure
+[2](reduceByFile_flow.png)) split the problem by file. Files are sent to
````
Make the same changes as in figure 1.

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,727 @@
+output of all MAP calls. `REDUCE` usually plays a minor role by concatenating or
+unlisting results.
+
+```{r echo = FALSE, fig2, fig.cap=": Mechanics of `reduceByFile` and `reduceFiles`"}
````
Make same changes as in figure 1

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,727 @@
+compare or combine data subsets across files. The first example in this section
+computes pileups on subsets from individual files then sums over all files. The
+second example computes coverage on a group of ranges for each file then
+performs a basepair-level t-test across files. The t-test example also
```
When I look at `t-test` in the original, the `t` looks like it has special formatting. Please change all instances of `t-test` to be formatted as

    `t`-test

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,727 @@
+tiles <- tileGenome(chr14_seqlen, ntile=5)
+```
+
+tiles is a `GRangesList` of length `ntile` with one range per element
````
Tiles should have backticks. In the PDF, GRangesList doesn't have special formatting.

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,727 @@
+## Records in a file
+
+`reduceByYield` iterates through records in a single file that would
+otherwise not fit in memory.It is similar to a one dimensional
```
Need a space after comma and before `It`

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,727 @@
+
+REDUCE combines the coverage vectors into a matrix, identifies all-zero
+rows, and performs row-wise t-testing using the `rowttests` function from
+the `r Biocpkg("genefilter")`  package. The index of which rows correspond to which basepair of the original range is stored as a column `offset`.
```
Wrap long line.

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,727 @@
+returns a chunk of data to work on and `DONE` is a function that defines a
+stopping criteria.
+
+Records from a single file are read by `readGAlignments` and limited by the `yieldSize` set in the BamFile.
```
Wrap long line

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,727 @@
+}
+```
+
+`bplapply` distributes the files to workers. Each worker uses `reduceByYield` to iteratively count and reduce overlaps in a BAM file.
````
Wrap long line

jwokaty commented:
I've requested additional edits. There are 2 pieces of the review left: `R CMD build` (it's always last) and checking that the .Rmd is representative of the .Rnw, which takes time because it is done manually at this point. Keep up the good work, I am sure that we will merge it this week.

sonali8434 committed:
Update GenomicFiles.Rmd

Made all the required changes.

sonali8434 committed:
Update GenomicFiles.Rmd

Added result="hide" in place of message =FALSE

sonali8434 committed:
Update GenomicFiles.Rmd

added message = FALSE

sonali8434 committed:
Update GenomicFiles.Rmd

sonali8434 commented:
@jwokaty I have done all the changes needed and also checked that that the .Rmd is representative of the .Rnw and done 2 changes rest everything is fine you can review it and we can proceed further. 
Thank you.

sonali8434 committed:
Update GenomicFiles.Rmd

wrapped two lines .

sonali8434 commented:
@jwokaty  Please review my PR I have checked R CMD build also after making all the changes.
Thanku..

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,738 @@
+data from the same worker, not across workers.
+
+The chart in Figure 
+<a id="fig1" ></a>
```
Move `<a id="fig1" ></a>` to just before line 158. How it should work is you click on the link `1` and it should take you to where the image is in the page.

sonali8434 replied:
yes it is working fine did you check it after knit?

sonali8434 replied:
Because actually I have checked it myself before and it is working super fine.

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,738 @@
+```
+
+In contrast to the 'byRange' approach, `reduceByFile` and `reduceFiles` (Figure
+<a id="fig2" ></a>
````
Move this line to just before line 173.

sonali8434 replied:
@jwokaty links are are working fine in the html document do we still need to change its its position? please clarify.
https://rpubs.com/Sonali8434/1014072

sonali8434 replied:
I have publish this just now.

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,738 @@
+
+```{r pileups-ranges}
+gr <- GRanges("chr14", IRanges(c(19411677, 19659063, 105421963,
+     105613740), width=20))
````
Usually, with R if the arguments span more than one line, we try to line it up with the start of the arguments. This is what they follow in the PDF for this section.

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,738 @@
+     105613740), width=20))
+```
+
+The `bam2R` function from the deepSNV package is used to compute the
````
deepSNV is a Bioconductor package. Please use `Biocpkg` to generate the link here and other references to deepSNV below.

sonali8434 replied:
ohk

sonali8434 replied:
In pdf we don't have link to all the all the references I have seen if we put a reference link for one package only once.. so for deepSNV also I think we need to make a reference only once.. am I correct please correct me ..

jwokaty replied:
There are no links in the PDF, but we make them for the HTML as a convenience for the users. There is one more reference to deepSNV at line 207.

sonali8434 replied:
ohk clear now i am asking because i don't want to delay the process of review further .
Thank you so much for clarification.

jwokaty commented on vignettes/GenomicFiles.Rmd:
```diff
@@ -0,0 +1,738 @@
+'tiles' or ranges that span the whole file. The `tileGenome` function creates a
+set of continuous ranges that span a given seqlength(s). The sample BAM files
+contain only chr14 so we extract the appropriate seqlength from the BAM files
+and use it in`tileGenome`. In this example we create 5 ranges but the optimal
```
Add a space between `in` and the starting backtick surrounding `tileGenome`.

jwokaty commented on vignettes/GenomicFiles.Rmd:
````diff
@@ -0,0 +1,738 @@
+# `sessionInfo()`
+
+```{r sessionInfo, echo=FALSE}
+(sessionInfo())
````
Remove parenthesis surrounding `sessionInfo()`

sonali8434 replied:
ohk

jwokaty commented:
@sonali8434 Thank you for this PR. It was not an easy first vignette and it was a long medium conversion. Please finish the remaining requests. If nothing else needs to be modified after you push your final changes, I will merge your PR. You're welcome to choose another vignette. I recommend choosing a difficult one.

sonali8434 committed:
Update GenomicFiles.Rmd

Made all the required changes.

sonali8434 commented:
@jwokaty Thank you so much for the review I have done all the changes after your clarification. 
under your guidance I have learned so much in this vignette conversion. Please make a review and do let me know if any further changes required.

jwokaty committed:
Fixed formatting

jwokaty committed:
Fixed formatting for pileups-ranges

jwokaty commented:
@sonali8434 Thanks again for your contribution. Your PR has been merged. I made a few formatting changes that you can see in https://github.com/Bioconductor/GenomicFiles/pull/8/commits/4d242b92a5a049b21bb172ae2e3543229a23b287 and https://github.com/Bioconductor/GenomicFiles/pull/8/commits/3acbb0a67ee5847dc545009cdab75f3b648c0c00.

sonali8434 commented:
> @sonali8434 Thanks again for your contribution. Your PR has been merged. I made a few formatting changes that you can see in [4d242b9](https://github.com/Bioconductor/GenomicFiles/commit/4d242b92a5a049b21bb172ae2e3543229a23b287) and [3acbb0a](https://github.com/Bioconductor/GenomicFiles/commit/3acbb0a67ee5847dc545009cdab75f3b648c0c00).

Thank you so much  @jwokaty. I have well noted the changes.
